api-spec/protobuf/ark/v1/service.proto

```diff
@@ -119,6 +125,12 @@ message RegisterIntentResponse {
   string request_id = 1;
 }
 
+message DeleteIntentRequest {
+  string request_id = 1;
```

Comment:
these should be oneOf. EIther you provide the intentid (which is a user secret) , or you provide an input (a single input is enough) in a bip322. The reasoning is that intent id is trivial and straightforward, but in the case of a wallet recovery where you only have a private key, the bip322 option works.